### PR TITLE
Using MissingOptionalLibraryError for consistency

### PR DESCRIPTION
**Pull request by 1ucian0**

### CODE REVIEW

**Review by mtreinish — COMMENTED** [review]
This will be a breaking change and something we can do, because anywhere someone was catching the importerror before would no longer work. That is unless missing optional library inherits from importerror which I don't think it does.

**Comment by 1ucian0** [comment]
> That is unless missing optional library inherits from importerror which I don't think it does.

It doesnt. I can add it.

More important, I'm more interested in the consistency than in the way. Do you think it makes more sense to turn those `MissingOptionalLibraryError` into `ImportError`?

**Comment by Cryoris** [comment]
So in my opinion the `MissingOptionalLibraryError` forces us to have proper errors that tell us 1. why it was raised, 2. how the package is called and 3. how to install it. I think that's helpful for users so I'd be in favor of the `MissingOptionalLibraryError` as opposed to (possibly lazily written) `ImportError`s 🙂

**Comment by mtreinish** [comment]
> > That is unless missing optional library inherits from importerror which I don't think it does.
> 
> It doesnt. I can add it.
> 
> More important, I'm more interested in the consistency than in the way. Do you think it makes more sense to turn those `MissingOptionalLibraryError` into `ImportError`?

Yeah that would be fine, if we make `MissingOptionalLibraryError` a subclass of `ImportError` it would resolve the backwards compatibility concern. I also think it should be a permanent change so we are consistently returning the same exceptions moving forward.

**Comment by 1ucian0** [comment]
I added the subclassing and finished with the rest of the imports. It's ready for review!

File: qiskit/transpiler/coupling.py [hunk 1]
```diff
@@ -334,15 +335,17 @@ def draw(self):
...
         try:
             import pydot
             from PIL import Image
         except ImportError as ex:
-            raise ImportError(
-                "CouplingMap.draw requires pydot and pillow. " "Run 'pip install pydot pillow'."
+            raise MissingOptionalLibraryError(
+                libname="pillow",
+                name="coupling map drawer",
+                pip_install="pip install pillow",
```
Comment (Cryoris) [inline qiskit/transpiler/coupling.py:348]: Should this be 
```suggestion
                pip_install="pip install pydot pillow",
```

File: qiskit/visualization/pass_manager_visualization.py [hunk 2]
```diff
@@ -98,19 +99,27 @@ def pass_manager_drawer(pass_manager, filename=None, style=None, raw=False):
 
     HAS_GRAPHVIZ = has_graphviz  # pylint: disable=invalid-name
 
+    if not HAS_GRAPHVIZ:
```
Comment (Cryoris) [inline qiskit/visualization/pass_manager_visualization.py:102]: It looks like we're checking twice for graphviz, is this intended?

**Review by Cryoris — COMMENTED** [review]
Two minor comments/questions otherwise LGTM

File: qiskit/transpiler/coupling.py (hunk 1 above)
Comment (1ucian0) [reply qiskit/transpiler/coupling.py:355]: Good catch! I unfold that one in eb4de93780d90400df0f07c6d4548e295247a289

File: qiskit/visualization/pass_manager_visualization.py (hunk 2 above)
Comment (1ucian0) [reply qiskit/visualization/pass_manager_visualization.py:102]: not intended indeed! One of the checks is being removed in 4b46f1450

**Review by Cryoris — APPROVED** [review]

**Pull request merged**